src/autopas/containers/verletClusterLists/VerletClusterCellsParticleIterator.h

```diff
@@ -195,20 +206,40 @@ class VerletClusterCellsRegionParticleIterator
    */
   explicit VerletClusterCellsRegionParticleIterator(CellVecType *cont, DummyStartsType &dummyStarts,
                                                     std::array<double, 3> startRegion, std::array<double, 3> endRegion,
-                                                    const std::vector<size_t> &indicesInRegion, double offsetToDummy,
+                                                    std::vector<size_t> indicesInRegion, double offsetToDummy,
                                                     IteratorBehavior behavior = haloAndOwned, double skin = 0.0)
       : VerletClusterCellsParticleIterator<Particle, ParticleCell, modifiable>(cont, dummyStarts, offsetToDummy,
                                                                                behavior),
         _startRegion(startRegion),
         _endRegion(endRegion),
-        _indicesInRegion(indicesInRegion),
+        _indicesInRegion(std::move(indicesInRegion)),
         _currentRegionIndex(0),
         _skin(skin) {
+    // 1. add a last entry within _indicesInRegion that points beyond the end of _vectorOfCells.
     _indicesInRegion.push_back(this->_vectorOfCells->size() + 1);
+
+    // 2. set _currentRegionIndex to current thread id
+    // also ensure that _currentIndex does not go beyond _indicesInRegion.
+    _currentRegionIndex = std::min(static_cast<unsigned long>(autopas_get_thread_num()), _indicesInRegion.size() - 1);
+
+    // 3. set the _cellID to the appropriate value
     this->_cellId = _indicesInRegion[_currentRegionIndex];
-    this->_cellIter = (*this->_vectorOfCells)[this->_cellId]._particles.begin();
-    this->_cellEnd = this->_cellIter + this->getDummyStartbyIndex(this->_cellId);
-    --this->_cellIter;
+
+    if (this->_cellId >= this->_vectorOfCells->size()) {
+      // prevent segfaults if the _cellId is too large.
+      return;
+    }
+
+    // 4. set _iteratorWithinOneCell to begin of that cell
+    this->_iteratorWithinOneCell = (*this->_vectorOfCells)[this->_cellId]._particles.begin();
+
+    // 5. adapt _cellEnd to dummies.
+    this->_cellEnd = this->_iteratorWithinOneCell + this->getDummyStartbyIndex(this->_cellId);
+
+    // 6. do a -- to be able to do a ++.
+    --this->_iteratorWithinOneCell;
```

Comment:
the way it is documented here it sounds useless so maybe clarify why you want to do the ++ ;)